components/arm/collectors_test.go

```diff
@@ -78,7 +79,12 @@ func TestCollectors(t *testing.T) {
 			col.Collect()
 			mockClock.Add(captureInterval)
 
-			test.That(t, buf.Length(), test.ShouldEqual, 1)
+			length := 0
+			for i := 0; i < numRetries && length == 0; i++ {
+				length = buf.Length()
+				time.Sleep(time.Second)
+			}
+			test.That(t, length, test.ShouldBeGreaterThan, 0)
```

Comment:
Can we check if length != 0 and break the loop if so? That way we don't have to sleep if it's not necessary

Reply:
The && on the loop should do this already. If length != 0, then it will break out of the loop. Granted, it will sleep at least once everytime. I can change that if you'd like.

Reply:
Yeah I meant so we didn't have to sleep at least once - given that this has only failed once, the sleep is usually not necessary!

Reply:
Oh yea I gotcha. With the change to a helper, I'm including that so it should still pass really fast ~<1 second